torch/distributed/_composable/contract.py

```diff
@@ -116,19 +92,16 @@ def contract(
     # wraps will make functions decorated with contract() pickleable - needed for integration with torch.package
     @wraps(state_cls)  # type: ignore[arg-type]
     def inner(
-        func: Callable[
-            Concatenate[Union[nn.Module, Sequence[nn.Module]], _P], Optional[nn.Module]
-        ]
-    ) -> _ContractFn[
-        Concatenate[Union[nn.Module, Sequence[nn.Module]], _P], _T, _TState
-    ]:
+        func: Callable[Concatenate[_M, _P], _M]
+    ) -> _ContractFn[Concatenate[_M, _P], _M, _TState]:
         @wraps(func)
         def wrapper(
-            module: Union[nn.Module, Sequence[nn.Module]],
+            module: _M,
             *args: _P.args,
             **kwargs: _P.kwargs,
-        ) -> Optional[nn.Module]:
+        ) -> _M:
```

Comment:
Something I missed on my first read through. Shouldn't this actually be Optional[_M]?

Reply:
@Skylion007 I believe it's not supposed to return None: https://github.com/pytorch/pytorch/blob/4ea9da692ad60bbe3d2516c1774c4c181a4e7b62/torch/distributed/_composable/contract.py#L152-L154